battle-service/src/main/java/org/astashonok/battleservice/factories/impl/BattleFactoryImpl.java

```diff
@@ -0,0 +1,48 @@
+package org.astashonok.battleservice.factories.impl;
+
+import org.astashonok.battleservice.entities.Battle;
+import org.astashonok.battleservice.exceptions.BattleCreationFormException;
+import org.astashonok.battleservice.factories.BattleFactory;
+import org.astashonok.battleservice.models.BattleCreationForm;
+import org.astashonok.battleservice.services.AuthUserService;
+import org.astashonok.battleservice.validators.BattleCreationFormValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class BattleFactoryImpl implements BattleFactory {
+
+    @Autowired
```

Comment:
it's metter of taste but i would  like to use contructor injection. so you can mark your fields as final and use  Lombok''s RequiredArgsConstructor annotation